### PR TITLE
Update version of AMQP dependency

### PR DESCRIPTION
**Pull request by JoshLove-msft**
https://github.com/azure/azure-sdk-for-net/issues/12624

### CODE REVIEW

File: eng/Packages.Data.props [hunk 1]
```diff
@@ -29,7 +29,7 @@
     <PackageReference Update="Castle.Core" Version="4.4.0" />
     <PackageReference Update="FluentAssertions" Version="5.10.3" />
     <PackageReference Update="FsCheck.Xunit" Version="2.14.0" />
-    <PackageReference Update="Microsoft.Azure.Amqp" Version="[2.4.2, 3.0.0)" />
+    <PackageReference Update="Microsoft.Azure.Amqp" Version="2.4.4" />
```
Comment (JoshLove-msft) [inline eng/Packages.Data.props:32]: Removing the 3.0.0 max version as there are no longer plans to introduce breaks in this version.

**Review by jsquire — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

**Pull request merged**